src/rules/if_change_then_change.rs

```diff
@@ -11,54 +12,103 @@ use regex::Regex;
 use crate::diagnostic;
 use crate::git;
 
+#[derive(Debug)]
+pub enum ThenChange {
+    RemoteFile(String),
+    RepoFile(PathBuf),
+    MissingIf,
+    MissingThen,
+    None,
+}
+
 #[derive(Debug)]
 pub struct IctcBlock {
     pub path: PathBuf,
     pub begin: i64,
     pub end: i64,
-    pub thenchange: PathBuf,
+    pub thenchange: ThenChange,
 }
 
-lazy_static::lazy_static! {
-    static ref RE_BEGIN: Regex = Regex::new(r"(?i)^\s*(//|#)\s*ifchange\s*$").unwrap();
-    static ref RE_END: Regex = Regex::new(r"(?i)^\s*(//|#)\s*thenchange\s*(.*)$").unwrap();
+impl IctcBlock {
+    fn get_range(&self) -> diagnostic::Range {
+        diagnostic::Range {
+            path: self.path.to_str().unwrap().to_string(),
+            start: diagnostic::Position {
+                line: self.begin as u64,
+                character: 0,
+            },
+            end: diagnostic::Position {
+                line: self.end as u64,
+                character: 0,
+            },
+        }
+    }
+}
 
+lazy_static::lazy_static! {
+    static ref RE_BEGIN: Regex = Regex::new(r"(?i)^\s*(//|#)\s*ifchange(.*)$").unwrap();
```

Comment:
Any reason to capture the content after `ifchange`—`(.*)$`? If we aren't using it (we just check `.is_some()` below, then we should remove that part of this regex

Reply:
Yeah - in the next PR I am going to use that content to point to a remote location so something like

IfChange github.com/trunk-io/trunk/file.cpp@{commit}

Reply:
That's neat 👍